xoto3/dynamodb/write_versioned/api2.py

```diff
@@ -168,3 +176,20 @@ def create_or_update_trans(vt: VersionedTransaction) -> VersionedTransaction:
         return table.put(creator_updater(table.get(key)(vt)))(vt)
 
     return create_or_update_trans
+
+
+def write_item(
```

Comment:
I wonder if it would be surprising to someone to see that a function called `write_item` will delete the item if the `updater` returns `None`. 
I'm trying to think if there is a better name, but I can't really think of one. `put_or_delete_item` maybe. That's a lot longer than what you have, though.

Maybe just something in the docstring that calls that out would be good enough?

Reply:
Just playing devil's advocate but wouldn't you still technically be writing the item to the database just doing a soft delete?

Reply:
actually, this is a super important point to bring up.

DynamoDB explicitly uses the word `write` to describe the set of "both puts and deletes", as you can see here and elsewhere. https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html

I think it's best that we try to capture their terminology with ours even in these higher level abstractions. I've tried to honor their terms in the past, and this is just an extension of that. 

I agree that delete isn't what I necessarily think of first when I see the word write, but ultimately I think it's best to hew closer toward what the words mean in the underlying system, rather than inventing my own subtly different meanings for their own words.

Reply:
Crazy. This wasn't here when I wrote my comment, very much along the same lines.

_@jstephensxoi -- no, this would do a hard delete, since a soft delete requires an update to the record to reflect the deleted_at and deleted_by values..._

Reply:
(i think you may have experienced the weirdness of git force-push here, Nathan. it definitely wasn't there before. :) )